package exercises;

import java.util.Random;
import java.util.concurrent.Callable;
import java.util.concurrent.ExecutionException;
import java.util.concurrent.ExecutorService;
import java.util.concurrent.Executors;
import java.util.concurrent.Future;

public class CallApp {

	public static void main(String[] args) {
		ExecutorService executor = Executors.newCachedThreadPool();
				
		System.out.println("Starting...");
		
		Future<Integer> future = executor.submit(new Callable<Integer>() {
			public Integer call() {
				
				Random random = new Random();
				int duration = random.nextInt(4000);
				
				try {
					Thread.sleep(duration);
				} catch (InterruptedException e) {
					// TODO Auto-generated catch block
					e.printStackTrace();
				}
				System.out.println("Finished...");
				return duration;
			}
				
			
		});
		
		executor.shutdown();
		//executor.awaitTermination(timeout, unit)
		try {
			System.out.println("Result is: " + future.get());
		} catch (InterruptedException | ExecutionException e) {
			// TODO Auto-generated catch block
			System.out.println(e);
		}
	}

}
